src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -142,6 +146,14 @@ public boolean deleteFile(FireStoreFile fireStoreFile) {
     BlobCrl blobCrl = getBlobCrl(destinationClientFactory);
     try {
       blobCrl.deleteBlob(blobName);
+      String folderPath = blobName.split("/")[0];
+      try {
+        // Attempt to delete the file's folder
+        blobCrl.deleteBlob(folderPath);
+      } catch (BlobStorageException e) {
+        // Attempt to delete the parent but this should not cause the overall failure of the file
+        logger.warn("Could not delete the blob folder {}", folderPath, e);
+      }
```

Comment:
I don't totally understand this - why would we want to delete the file's folder on an individual file delete? Could there ever be more than one file in the folder?

Reply:
Great question.  The way the files are laid out is like in GCP as:
`fileId/sourceFileName`

So there's only ever one file per directly.  Because we're using ADLS gen2 with hierarchical storage enable, we have directories as a thing.  These need to be explicitly deleted (as opposed to the standard blob store format where there are no directories...just a giant list of files

Reply:
Ok, thanks for explaining!

Reply:
My worry about code like this is race conditions involving two different threads writing to the same location at the same time, e.g. if we delete while someone else is writing. As long as creating the directory isn't a separate step (that is, it's done internally by azure) then it's probably not an issue, as I'd expect azure to perform these operations atomically at the API level.